WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserFragment.kt

```diff
@@ -0,0 +1,433 @@
+        val header = inflater.inflate(R.layout.theme_grid_cardview_header, null)
+
+        // inflater doesn't work with automatic elevation in night mode so we set card background color manually
+        val headerCardView = header.findViewById<View>(R.id.header_card)
+        val elevationOverlayProvider = ElevationOverlayProvider(header.context)
+        val elevatedSurfaceColor =
+            elevationOverlayProvider.compositeOverlayWithThemeSurfaceColorIfNeeded(headerCardView.elevation)
+        headerCardView.setBackgroundColor(elevatedSurfaceColor)
+
+        currentThemeTextView = header.findViewById(R.id.header_theme_text)
+
+        setThemeNameIfAlreadyAvailable()
+        headerCustomizeButton = header.findViewById(R.id.customize)
+        headerCustomizeButton?.setOnClickListener {
+            AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.THEMES_CUSTOMIZE_ACCESSED, site)
+            callback?.onTryAndCustomizeSelected(currentThemeId)
+        }
+
+        val details = header.findViewById<LinearLayout>(R.id.details)
+        details.setOnClickListener { callback?.onDetailsSelected(currentThemeId) }
+
+        val support = header.findViewById<LinearLayout>(R.id.support)
+        support.setOnClickListener { callback?.onSupportSelected(currentThemeId) }
+
+        binding.themeListview.addHeaderView(header)
+    }
+
+    private fun setThemeNameIfAlreadyAvailable() {
+        val currentTheme = themeStore.getActiveThemeForSite(requireNotNull(site))
+        currentTheme?.let { currentThemeTextView?.text = it.name }
+    }
+
+    fun setRefreshing(refreshing: Boolean) {
+        shouldRefreshOnStart = refreshing
+        swipeToRefreshHelper?.let {
+            it.isRefreshing = refreshing
+            if (!refreshing) {
+                refreshView()
+            }
+        }
+    }
+
+    private fun updateDisplay() {
+        if (!isAdded || view == null) {
+            return
+        }
+
+        val hasThemes = adapter.unfilteredCount > 0
+        val hasVisibleThemes = adapter.count > 0
+        val hasNoMatchingThemes = hasThemes && !hasVisibleThemes
+
+        binding.emptyView.visibility = if (!hasThemes) View.VISIBLE else View.GONE
+        if (!hasThemes && !NetworkUtils.isNetworkAvailable(activity)) {
+            binding.textEmpty.setText(R.string.no_network_title)
+        }
+        binding.themeListview.visibility = if (hasVisibleThemes) View.VISIBLE else View.GONE
+        binding.actionableEmptyView.visibility = if (hasNoMatchingThemes) View.VISIBLE else View.GONE
+    }
+
+    private fun fetchThemes(): List<ThemeModel> {
+        site?.let {
+            return if (it.isWPCom) {
+                sortedWpComThemes
+            } else {
+                sortedJetpackThemes
+            }
+        }
+
+        return ArrayList()
+    }
+
+    fun refreshView() {
+        adapter.setThemeList(fetchThemes())
+    }
+
+    private fun moveActiveThemeToFront(themes: MutableList<ThemeModel>?) {
```

Comment:
🤔 `ThemeStore.getThemesForSite(...)` returns its result from `WellSql`, and nullability is not handled at that layer. So, the result can be null. I see FluxC and WordPress-Android as separate projects. Unless FluxC explicitly states that the result is non-null, it's better to expect nullable values from FluxC functions.
[ThemeSqlUtils.getThemesForSite()](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/trunk/fluxc/src/main/java/org/wordpress/android/fluxc/persistence/ThemeSqlUtils.java#L104-L109) returns WellSql query result, and I don't know if it returns empty lists or null. Even if it does not currently return null, it might return null in future versions. So, I suppose the safer approach is keeping these nullable. 

> FYI: You end-up concluding to a similar this for ThemeStore.getWpComThemes() -> ThemeSqlUtils.getWpComThemes(), did you? 

Yes, the situation is the same for `ThemeStore.getWpComThemes()`. But `ThemeBrowserFragment` expects non-null values from it. So, I didn't want to change that behavior in this PR.

Since you have implemented your recommended suggestion throughout the entire repo, I can follow the same here too. However, I would appreciate your input on my viewpoint.

Reply:
Thanks for sharing your viewpoint @irfano , a total valid one! 💯 

However, from the looks of it, and even though FluxC is not defining if a method is returning a non-null or nullable object (something expected), when that returned object is a list, I have "safely" assumed that this list is empty, never `null`, thus me suggesting this change to you. 😊

I came to this realization after lots of interactions with FluxC on nullability annotation work, thus I am 99.99% sure that a list returned from FluxC is never going to be null. My bet on this is also due to the fact that if you searched throughout the codebase you will notice that almost every logic/code we already have is depending on that, list not being return as null from a query. 🤷

Reply:
Thank you for explaining it. 

> I came to this realization after lots of interactions with FluxC on nullability annotation work, thus I am 99.99% sure that a list returned from FluxC is never going to be null.

Good to know this. 👍🏻 

I believe it's best to maintain consistency, and we will make any future changes in FluxC, so, I have followed your suggestion: 537c6e37377777f430ab63ce0e98b9ec67bb38b9

Reply:
@irfano I think this discussion here is similar to [that one](https://github.com/wordpress-mobile/WordPress-Android/pull/19494#discussion_r1378906067). As such, I am resolving this too so we can move forward. 💯 🙇 🌟